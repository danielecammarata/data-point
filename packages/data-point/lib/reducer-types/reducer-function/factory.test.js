/* eslint-env jest */

const factory = require("./factory");
const createReducer = require("../index").create;

test("reducer/reducer-function#isType", () => {
  expect(factory.isType("$foo")).toBe(false);
  expect(factory.isType("foo()")).toBe(false);
  expect(factory.isType(() => true)).toBe(true);
});

describe("reducer/reducer-function#validateFunction", () => {
  /* eslint-disable no-unused-vars */
  expect(factory.validateFunction(() => true)).toBe(true);
  expect(factory.validateFunction(value => true)).toBe(true);
  expect(factory.validateFunction((value, acc) => true)).toBe(true);
  expect(factory.validateFunction((value, acc, next) => true)).toBe(true);
  expect(() =>
    // 4 arguments is not a reducer
    factory.validateFunction((a, b, c, d) => true)
  ).toThrowErrorMatchingSnapshot();
  /* eslint-enable no-unused-vars */
});

describe("reducer/reducer-function#create", () => {
  test("function body", () => {
    const reducerFunction = () => value => value * 2;
    const reducer = factory.create(createReducer, reducerFunction);
    expect(reducer.body).toEqual(reducerFunction);
  });
});
