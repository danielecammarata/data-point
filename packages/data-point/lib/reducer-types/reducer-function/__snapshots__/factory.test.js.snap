// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `"Reducer functions must have an arity of 3 at most"`;
